marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java

```diff
@@ -213,51 +220,119 @@ public void deleteTopic(Long topicId) {
     topicRepository.deleteById(topicId);
   }
 
+    /**
+     * Searches through the topic repository and get all topics corresponding
+     * to requested search parameters: subCategory and requestOrOffer.
+     * Then the output list is filtered according to the given search values: from_size and to_price.
+     * This implementation ignores radius, as search value parameter.
+     *
+     * @param topicSearchRequestDTO has multiple search parameters. One of them is topic_values,
+     *                              which it is optional and contains search fields: from_size, to_price and radius.
+     *
+     * @return List of filtered
+     */
   public TopicSearchListResponseDTO searchTopic(TopicSearchRequestDTO topicSearchRequestDTO) {
-    TopicSearchListResponseDTO result = new TopicSearchListResponseDTO();
-
-    Topic topic = new Topic();
-    topic.setRequestOrOffer(topicSearchRequestDTO.getRequestOrOffer());
-    if (topicSearchRequestDTO.getSearchValues() != null) {
-      List<TopicValue> collect = topicSearchRequestDTO.getSearchValues().stream().map(value -> {
-        return TopicValue.builder().value(value.getValue()).build();
-      }).collect(Collectors.toList());
+        TopicSearchListResponseDTO result = new TopicSearchListResponseDTO();
+
+        List<TopicValue> topicValues = Collections.emptyList();
+        if (topicSearchRequestDTO.getSearchValues() != null) {
+            topicValues = topicSearchRequestDTO.getSearchValues().stream()
+                    .map(value -> TopicValue.builder()
+                            .value(value.getValue())
+                            .topicValueId(value.getFieldId())
```

Comment:
Hier bin ich am nachdenken, ob das so Sinn macht. Wir nehmen die field_id und setzen sie als topicValueId. Warum? Ist diese nicht automatisch generiert als primary key?
Gibt es evtl. `.fieldId(value.getFieldId()` ?

Reply:
Here I am creating a TopicValue object.
TopicValue has no "fieldId" property. It has only "topicValueId". That is why I can`t do ".fieldId(value.getFieldId())"